ultrasync/main.py

```diff
@@ -2152,16 +2151,16 @@ def set_output_control(self, output, state):
 
         At this time, this feature is only supported by the COMNAV panels
         """
+        if not self.session_id and not self.login():
```

Comment:
Why log in when the vendor is already known? This check should go back where it was; but i do see the mistake i made with the other code.  the rest looks okay.

Reply:
From my understanding, when the command `ultrasync -c config.txt --output 1 --switch 1` is called with the existing code, it always produces an error because the vendor is unknown. 

This is because when the function is called, it checks the vendor before it has logged into the alarm, so `self.vendor` is equal to `Unknown`. So from what I'm gathering, it needs to run the login() function before it checks the vendor type:
```
        if not self.session_id and not self.login():
            return False
```

I had a look at the `set_zone_bypass` function and this also checks the vendor type after the `login()` function is called. The `if` statement that gets the `session_id` and `login` is called at the start of the `set_zone_bypass` function and then `if` statements followed by an `else` statement for `ZEROWIRE` and `XGEN` alarms gives an error to display that the alarm isn't supported for zone bypass:
```
  else:   # self.vendor is NX595EVendor.{ZEROWIRE, XGEN}

            logger.error(
                'Bypass not implemented for vendor {}'.format(self.vendor))
            return False
```

So if the `set_zone_bypass` function is anything to go by, then the `set_output_control` should have a similar structure to this function, otherwise ultrasync would be unable to switch the outputs on and off.

Hopefully this make sense. If you have a better idea on how it should be structured to work then I'm also open to that.